public/ui/app.js

```diff
@@ -176,7 +176,7 @@ export default class App extends Component {
 					</div>
 
 					<div class="Hand">
-						<${Cards} cards=${state.hand} isHand=${true} energy=${state.player.currentEnergy} />
+						<${Cards} cards=${state.hand} isHand=${true} state=${state} />
```

Comment:
It's a bit confusing to have a prop named `state`. Then we have `this.props.state` vs `this.state` which are notthe same thing. Maybe rename to `gameState`?